lib/Controller/ViewController.php

```diff
@@ -148,4 +156,37 @@ private function getView(string $view): string {
 				return $view;
 		}
 	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * This function makes the colour dots work for mobile widgets
+	 *
+	 * Returns an SVG with size32x32 if the hex colour is valid
+	 *
+	 * @param string $color - url encoded HEX colour
+	 * @return FileDisplayResponse
+	 * @throws NotFoundException
+	 * @throws NotPermittedException
+	 */
+	public function getCalendarDotSvg(string $color = "#0082c9"): FileDisplayResponse {
+		if (preg_match('/^#([0-9a-f]{3}|[0-9a-f]{6})$/i', urldecode($color))) {
+			$svg = '<svg height="32" width="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><rect width="100%" height="100%" fill="' . urldecode($color) . '"/></svg>';
+		} elseif (preg_match('/^#([0-9a-f]{3}|[0-9a-f]{6})$/i', '#' . urldecode($color))) {
```

Comment:
I would only define `$validColor` in each condition branch, then define `$svg` only once below for clarity.

You could also have made the `#` in the regex optional, then `trim($color, '#')` and finally add the `#` in the svg template itself. That would save a `elseif` call, but probably not worth it in terms of performance.